storage-manager/src/KVStorageInitializer.cpp

```diff
@@ -0,0 +1,74 @@
+/* Copyright (C) 2024 MariaDB Corporation
+
+   This program is free software; you can redistribute it and/or
+   modify it under the terms of the GNU General Public License
+   as published by the Free Software Foundation; version 2 of
+   the License.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program; if not, write to the Free Software
+   Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston,
+   MA 02110-1301, USA. */
+#include "Config.h"
+#include "SMLogging.h"
+#include "KVStorageInitializer.h"
+
+namespace storagemanager
+{
+static std::shared_ptr<FDBCS::FDBDataBase> fdbDataBaseInstance;
+static std::unique_ptr<FDBCS::FDBNetwork> fdbNetworkInstance;
+static std::mutex kvStorageLock;
```

Comment:
I'd better avoid static mutexes, we have a plenty of code, that starts before main() and we could have undefined statics initialization effects. Please consider explicit initialization inside of class object

Reply:
@mariadb-LeonidFedorov  The static here means that particular global is not visible outside the translation unit. The idea behind this globals is to implement a singleton pattern, unfortunately we cannot initialize fdb network more than one time, but without a global variable you cannot say is it already init or not. I'm not sure which problem with static initialization you are talking inititialization order fiasco is not applicable here. If you know how to share information across threads without global, please let me know.